In [1]:
import numpy as np
import pandas as pd

In [2]:
df = pd.read_excel('Datos_Compañia_Seguros.xlsx')

In [3]:
headers = df.iloc[0]
df  = pd.DataFrame(df.values[1:], columns=headers)

In [4]:
# Number of unique identifiers.
len(df[df.NUM_ID.notnull()].NUM_ID.unique())

957

In [5]:
# Number of clients.
df.shape[0]

960

In [6]:
# Null unique identifiers.
df[df.NUM_ID.isnull()]

0  Servicio contratado FECHA_CONTRATACION   NOMBRE APELLIDO1 APELLIDO2  \
9            SEGURO_01                NaT  BEATRIZ       NaN       NaN   
10           SEGURO_04                NaT    JESUS       NaN       NaN   
11           SEGURO_09                NaT   CARLOS       NaN       NaN   

0   NAC_FECHA NUM_ID TIPO_ID DOM_SERVICIO_PROVINCIA DOM_SERVICIO_CP DOM_PAIS  \
9  1993-05-12    NaN     DNI                 MALAGA              08       ES   
10 1996-04-20    NaN     DNI                GRANADA              08       ES   
11 1991-09-19    NaN     DNI                GRANADA              08       ES   

0       TELF1 TELF2                           EMAIL  
9   647477620   NaN  beatriz.gomez.agirre@gmail.com  
10  685918989   NaN  jesus.hernandez.diaz@gmail.com  
11  673418203   NaN   carlos.torres.parra@gmail.com

There are 3 customers without an identifier. All other clients have a unique identifier.

In [7]:
# Clients without last name1 nor or without last name2.
df[(df.APELLIDO1.isnull()) | (df.APELLIDO2.isnull())]

0  Servicio contratado FECHA_CONTRATACION         NOMBRE  APELLIDO1 APELLIDO2  \
9            SEGURO_01                NaT        BEATRIZ        NaN       NaN   
10           SEGURO_04                NaT          JESUS        NaN       NaN   
11           SEGURO_09                NaT         CARLOS        NaN       NaN   
13           SEGURO_12                NaT         DANIEL        NaN       NaN   
14           SEGURO_02                NaT         RAFAEL        NaN       NaN   
17           SEGURO_05                NaT          SERGI  HERNANDEZ       NaN   
18           SEGURO_06                NaT       SANTIAGO     TORRES       NaN   
19           SEGURO_07                NaT           IVAN     MARTIN       NaN   
20           SEGURO_04                NaT         AMPARO    PIÑEIRO       NaN   
21           SEGURO_09                NaT    MARIA LUISA   GONZALEZ       NaN   
22           SEGURO_10                NaT    JUAN CARLOS     MARTIN       NaN   
23           SEGURO_11         2004-05-01  MARIA DOLORES  DOMINGUEZ       NaN   
25           SEGURO_02         2005-05-01          ISAAC    RAMIREZ       NaN   
26           SEGURO_03         2005-05-01          JORGE     ROBLES       NaN   
27           SEGURO_12         2006-05-01         RAQUEL  DOMINGUEZ       NaN   
28           SEGURO_04         1993-05-01           IVAN     GARCIA       NaN   
30           SEGURO_10         2005-05-01        ROSARIO     ORTEGA       NaN   
32           SEGURO_03         2006-05-01    JUAN CARLOS     GARCIA       NaN   
33           SEGURO_07         2006-05-01         SERGIO    CARDONA       NaN   
34           SEGURO_01         2006-05-01           SARA     LOZANO       NaN   
35           SEGURO_04         2007-05-01          PEDRO    RAMIREZ       NaN   

0   NAC_FECHA    NUM_ID TIPO_ID DOM_SERVICIO_PROVINCIA DOM_SERVICIO_CP  \
9  1993-05-12       NaN     DNI                 MALAGA              08   
10 1996-04-20       NaN     DNI                GRANADA              08   
11 1991-09-19       NaN     DNI                GRANADA              08   
13 1992-05-19  45390239     DNI                GRANADA              08   
14 1993-02-02  84747702     DNI  MADRID (COMUNIDAD DE)              28   
17 1997-02-04  85022402     DNI                GRANADA              08   
18 1992-05-25  82675376     DNI                GRANADA              08   
19 1995-01-25  40546977     DNI                GRANADA              08   
20 1993-06-19  39621305     DNI                GRANADA              08   
21 1996-05-08  52910253     DNI                 MALAGA              08   
22 1996-06-22  26361148     DNI                GRANADA              08   
23 1993-12-08  63888659     DNI                GRANADA              08   
25 1992-11-12  17961949     DNI                GRANADA              08   
26 1995-01-18  77394278     DNI                GRANADA              08   
27 1995-07-09  50525577     DNI                GRANADA              08   
28 1994-04-14  37122848     DNI                CÓRDOBA              14   
30 1993-04-04  41624754     DNI                GRANADA              08   
32 1992-01-10  20039527     DNI                GRANADA              08   
33 1997-02-23  89408635     DNI                 MALAGA              08   
34 1996-08-11  75987051     DNI                GRANADA              08   
35 1994-09-08  58628967     DNI                GRANADA              08   

0  DOM_PAIS      TELF1 TELF2                                   EMAIL  
9        ES  647477620   NaN          beatriz.gomez.agirre@gmail.com  
10       ES  685918989   NaN          jesus.hernandez.diaz@gmail.com  
11       ES  673418203   NaN           carlos.torres.parra@gmail.com  
13       ES  695412798   NaN        daniel.piñeiro.herrera@gmail.com  
14       ES  645347696   NaN           rafael.gonzalez.mas@gmail.com  
17       ES  684918341   NaN        sergi.quesada.gonzalez@gmail.com  
18       ES  617935430   NaN      santiago.ramirez.navarro@gmail.com  
19       ES  6642268

In [8]:
# Number of clients without last name1 nor or without last name2.
clientes_sinapellido = df[(df.APELLIDO1.isnull()) | (df.APELLIDO2.isnull())]
print(f'There are {len(clientes_sinapellido)} customers without the second last name and 5 of them have no last name.')

There are 21 customers without the second last name and 5 of them have no last name.


In [9]:
# Classes of contracted service.
sorted(df[df['Servicio contratado'].notnull()]['Servicio contratado'].unique())

['SEGURO_01',
 'SEGURO_02',
 'SEGURO_03',
 'SEGURO_04',
 'SEGURO_05',
 'SEGURO_06',
 'SEGURO_07',
 'SEGURO_08',
 'SEGURO_09',
 'SEGURO_10',
 'SEGURO_11',
 'SEGURO_12']

In [10]:
# All customers without phone 1 do not have phone 2.
df[df.TELF1.isnull()]

0   Servicio contratado FECHA_CONTRATACION       NOMBRE APELLIDO1   APELLIDO2  \
351           SEGURO_03         2011-05-01       ALICIA    MOLINA       BRAVO   
352           SEGURO_12         2011-05-01     VICTORIA      PEÑA      ALONSO   
357           SEGURO_03         2012-05-01  ENCARNACION      DIAZ   RODRIGUEZ   
364           SEGURO_02         2012-05-01       CARLOS   ALVAREZ         GIL   
365           SEGURO_03         2012-05-01    ANA BELEN    FERRER  SANTAMARIA   
370           SEGURO_04         2012-05-01         INES   SEMPERE     MONTOYA   
380           SEGURO_10         2013-05-01      ROBERTO     PEREZ       RAMOS   
421           SEGURO_09         2013-05-01        CELIA      DIAZ      LOZANO   
478           SEGURO_04         2013-05-01       ADRIAN     NUÑEZ  SANTAMARIA   
638           SEGURO_12         2013-05-01    ANA BELEN     ORTIZ     ALVAREZ   
792           SEGURO_05         2013-05-01        PILAR      SANZ   FERNANDEZ   
836           SEGURO_09         2016-05-01    JUAN JOSE   FIDALGO       YAÑEZ   
872           SEGURO_10         2015-05-01        NURIA    BLANCO     BELTRAN   
896           SEGURO_09         2017-05-01      DOLORES   MARQUEZ       GOMEZ   

0    NAC_FECHA    NUM_ID    TIPO_ID DOM_SERVICIO_PROVINCIA DOM_SERVICIO_CP  \
351 1996-03-03  61070231        DNI                CÓRDOBA              14   
352 1996-03-28  65701451        DNI                ALMERIA              17   
357 1993-09-06  91509350        DNI                CORDOBA              17   
364 1993-07-17  40709160        DNI                ALMERIA              17   
365 1995-08-21  45372493        DNI                CORDOBA              17   
370 1991-02-22  50030473  PASAPORTE                ALMERIA              17   
380 1993-10-31  46302263        DNI                CÓRDOBA              14   
421 1995-08-13  41495722        DNI      VALENCIA/VALÈNCIA              46   
478 1993-07-27  68177607  PASAPORTE      VALENCIA/VALÈNCIA              46   
638 1991-01-22  70419529  PASAPORTE                GRANADA              18   
792 1993-10-23  18058034        DNI      VALENCIA/VALÈNCIA              46   
836 1996-11-03  80134943        DNI                GRANADA              18   
872 1993-03-19  88240446  PASAPORTE      VALENCIA/VALÈNCIA              46   
896 1992-09-27  40540618        DNI                GRANADA              18   

0   DOM_PAIS TELF1 TELF2                          EMAIL  
351       ES   NaN   NaN         alicia.ramos.ramos@com  
352       ES   NaN   NaN      victoria.perez.olivan@com  
357       ES   NaN   NaN  encarnacion.alvarez.anton@com  
364       ES   NaN   NaN    carlos.rodriguez.torres@com  
365       ES   NaN   NaN   anabelen.gutierrez.bello@com  
370       ES   NaN   NaN           ines.zabala.roca@com  
380       ES   NaN   NaN    roberto.arribas.delgado@com  
421       ES   NaN   NaN                            NaN  
478       ES   NaN   NaN                            NaN  
638       ES   NaN   NaN                            NaN  
792       ES   NaN   NaN                            NaN  
836       ES   NaN   NaN                            NaN  
872       ES   NaN   NaN                            NaN  
896       ES   NaN   NaN                            NaN

14 customers do not have phone 1 or phone 2.

In [11]:
# Non-null values per variable.
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 960 entries, 0 to 959
Data columns (total 14 columns):
 #   Column                  Non-Null Count  Dtype         
---  ------                  --------------  -----         
 0   Servicio contratado     900 non-null    object        
 1   FECHA_CONTRATACION      923 non-null    datetime64[ns]
 2   NOMBRE                  960 non-null    object        
 3   APELLIDO1               955 non-null    object        
 4   APELLIDO2               939 non-null    object        
 5   NAC_FECHA               943 non-null    datetime64[ns]
 6   NUM_ID                  957 non-null    object        
 7   TIPO_ID                 960 non-null    object        
 8   DOM_SERVICIO_PROVINCIA  960 non-null    object        
 9   DOM_SERVICIO_CP         960 non-null    object        
 10  DOM_PAIS                948 non-null    object        
 11  TELF1                   946 non-null    object        
 12  TELF2                   49 non-null     object    

60 clients do not have a contracted service but they do have a contracting date.

37 clients do not have a service contracting date and they do have a contracted service.

17 clients have no date of birth.

12 clients have no country.

14 clients do not have telephone 1 or 2.

7 clients do not have email.

In [12]:
# What type of email servers are there and if they exist.
emails = df.EMAIL.unique().tolist()
servidor_mail = []
for email in emails:
    try:
        if email.find('@') != -1 and email[email.find('@') + 1:] not in servidor_mail:
            servidor_mail.append(email[email.find('@') + 1:])
        elif email.find('@') == -1:
            print(f'wrong email: {email}')
    except:
        print(email)
servidor_mail

nan


['com', 'gmail.com']

There are two types of email providers: one is wrong 'com' and the other is correct gmail.

In [13]:
# For each unique value of mail server how many clients are there.
emails = df.EMAIL.unique().tolist()
com = 0
gmail = 0
for email in emails:
    try:
        if email.find('@') != -1:
            if email[email.find('@') + 1:] == 'com':
                com += 1
            else:
                gmail += 1
        elif email.find('@') == -1:
            print(f'wrong email: {email}')
    except:
        print(email)
print(f'com has {com} values and gmail has {gmail} values.')

nan
com has 63 values and gmail has 890 values.


7 clients do not have email, 63 have the wrong email and 890 have gmail.com email.

In [14]:
# Emails with the letter ñ do not exist.
count = 0
count_gmail = 0
for mail in df[df.EMAIL.notnull()].EMAIL.unique():
    if 'ñ' in mail:
        count += 1
        if 'gmail' in mail:
            count_gmail += 1
print(f'There are {count} emails with the letter ñ and {count_gmail} gmail emails with the letter ñ.')

There are 65 emails with the letter ñ and 60 gmail emails with the letter ñ.


In [15]:
# Impossible phone numbers1 (they do not have 9 digits or do not start with 6.)
for tlf in df[df.TELF1.notnull()].TELF1:
    str_tlf = str(tlf)
    if str_tlf[0] != '6' or len(str_tlf) != 9:
        print(tlf)

In [16]:
# Impossible phone numbers2 (they do not have 9 digits or do not start with 6.)
for tlf in df[df.TELF2.notnull()].TELF2:
    str_tlf = str(tlf)
    if str_tlf[0] != '6' or len(str_tlf) != 9:
        print(tlf)

There are no impossible phone numbers.

In [17]:
# The dates of birth appear to be correct.
df[df.NAC_FECHA.notnull()].NAC_FECHA.unique().min(), df[df.NAC_FECHA.notnull()].NAC_FECHA.unique().max()

(numpy.datetime64('1991-01-03T00:00:00.000000000'),
 numpy.datetime64('1997-02-27T00:00:00.000000000'))

There are no impossible dates of birth.

In [18]:
# The hire dates appear to be correct.
df[df.FECHA_CONTRATACION.notnull()].FECHA_CONTRATACION.unique().min(), df[df.FECHA_CONTRATACION.notnull()].FECHA_CONTRATACION.unique().max()

(numpy.datetime64('1990-05-01T00:00:00.000000000'),
 numpy.datetime64('2017-05-01T00:00:00.000000000'))

There are no impossible hire dates.

In [19]:
# The type of ID is DNI and passport.
df.TIPO_ID.unique()

array(['DNI', 'PASAPORTE'], dtype=object)

There are only two types of ID: DNI and passport.

In [20]:
# Correct typos: tildes.
df.DOM_SERVICIO_PROVINCIA.unique()

array(['ALMERIA', 'GRANADA', 'MALAGA', 'NAVARRA (COMUNIDAD FORAL DE)',
       'CASTELLÓN', 'MADRID (COMUNIDAD DE)', 'CÓRDOBA', 'HUESCA',
       'ALMERÍA', 'CORDOBA', 'SEGOVIA', 'SEVILLA', 'ALBACETE', 'BURGOS',
       'VALLADOLID', 'MÁLAGA', 'ÁLAVA', 'BALEARES (ISLAS)', 'RIOJA (LA)',
       'TERUEL', 'ZAMORA', 'SANTA CRUZ DE TENERIFE', 'VIZCAYA', 'LEÓN',
       'HUELVA', 'PALMAS (LAS)', 'LUGO', 'CORUÑA, A', 'PALENCIA',
       'ALICANTE', 'VALENCIA/VALÈNCIA', 'TOLEDO', 'GUIPÚZCOA', 'CÁDIZ',
       'CÁCERES', 'GUADALAJARA', 'BADAJOZ', 'CANTABRIA',
       'ASTURIAS (PRINCIPADO DE)', 'PONTEVEDRA', 'SALAMANCA'],
      dtype=object)

The typographical errors of DOM_SERVICIO_PROVINCIA.

In [21]:
# The existing countries are Spain and Italy.
df.DOM_PAIS.unique()

array(['ES', nan, 'IT'], dtype=object)

In [22]:
# As there are only Spanish provinces, there must be a failure in the country.
df[df.DOM_PAIS == 'IT']

0   Servicio contratado FECHA_CONTRATACION    NOMBRE APELLIDO1 APELLIDO2  \
211           SEGURO_09         2010-05-01    ADRIAN  GONZALEZ  MIRALLES   
212           SEGURO_10         2010-05-01  SANTIAGO       PAZ   GIMENEZ   

0    NAC_FECHA    NUM_ID TIPO_ID DOM_SERVICIO_PROVINCIA DOM_SERVICIO_CP  \
211 1991-09-23  41869732     DNI                GRANADA              08   
212 1993-12-14  25007560     DNI                 MALAGA              08   

0   DOM_PAIS      TELF1 TELF2                             EMAIL  
211       IT  678388097   NaN     adrian.gomez.blanco@gmail.com  
212       IT  689592141   NaN  santiago.jimenez.fraga@gmail.com

There is an error in the country. There are only clients from Spain, not from Italy.

In [23]:
# Fix data type errors: convert string to int.
df.DOM_SERVICIO_CP.unique()

array([17, '08', 31, 12, 28, 14, 22, 43, '04', 40, '02', 25, '09', 47, 29,
       '01', '07', 26, 44, 49, 38, 48, 24, 21, 35, 27, 15, 34, '03', 46,
       45, 18, 20, 11, 10, 19, '06', 39, 33, 36, 37], dtype=object)

In [24]:
# Existing unique cp services.
new_cp = []
for cp in df.DOM_SERVICIO_CP.unique():
    if type(cp) == str:
        new_cp.append(int(cp))
    else:
        new_cp.append(cp)
sorted(set(new_cp))

[1,
 2,
 3,
 4,
 6,
 7,
 8,
 9,
 10,
 11,
 12,
 14,
 15,
 17,
 18,
 19,
 20,
 21,
 22,
 24,
 25,
 26,
 27,
 28,
 29,
 31,
 33,
 34,
 35,
 36,
 37,
 38,
 39,
 40,
 43,
 44,
 45,
 46,
 47,
 48,
 49]

Correct typographical errors of DOM_SERVICIO_CP

In [25]:
# Correct typographical error (tildes) of provinces.
df.DOM_SERVICIO_PROVINCIA = df.DOM_SERVICIO_PROVINCIA.apply(lambda x:
                            'ALMERÍA' if x == 'ALMERIA'
                            else ( 'MÁLAGA' if x == 'MALAGA'
                            else ( 'CÓRDOBA' if x == 'CÓRDOBA'
                            else x)))

In [26]:
# Correct error from str to integer.
df.DOM_SERVICIO_CP = df.DOM_SERVICIO_CP.apply(lambda x: int(x) if type(x) == str else x)

In [27]:
# Provinces that share service cp.
cp_keys = {}
for cp in df.DOM_SERVICIO_CP:
    cp_keys[cp] = []
for index, row in df.iterrows():
    cp_keys[row['DOM_SERVICIO_CP']].append(row['DOM_SERVICIO_PROVINCIA'])

for cp in cp_keys:
    cp_keys[cp] = set(cp_keys[cp])
for key in cp_keys:
    if len(cp_keys[key]) > 1:
        print(key, cp_keys[key])

17 {'ALMERÍA', 'CORDOBA'}
8 {'GRANADA', 'MÁLAGA'}
43 {'SEVILLA', 'GRANADA'}


In [28]:
# There are no duplicate clients (name and surname) with more than one identifier.
df['cliente'] = df['NOMBRE'] + ' ' + df['APELLIDO1'] + ' ' + df['APELLIDO2']
df[df['cliente'].duplicated()]

0  Servicio contratado FECHA_CONTRATACION         NOMBRE  APELLIDO1 APELLIDO2  \
10           SEGURO_04                NaT          JESUS        NaN       NaN   
11           SEGURO_09                NaT         CARLOS        NaN       NaN   
13           SEGURO_12                NaT         DANIEL        NaN       NaN   
14           SEGURO_02                NaT         RAFAEL        NaN       NaN   
17           SEGURO_05                NaT          SERGI  HERNANDEZ       NaN   
18           SEGURO_06                NaT       SANTIAGO     TORRES       NaN   
19           SEGURO_07                NaT           IVAN     MARTIN       NaN   
20           SEGURO_04                NaT         AMPARO    PIÑEIRO       NaN   
21           SEGURO_09                NaT    MARIA LUISA   GONZALEZ       NaN   
22           SEGURO_10                NaT    JUAN CARLOS     MARTIN       NaN   
23           SEGURO_11         2004-05-01  MARIA DOLORES  DOMINGUEZ       NaN   
25           SEGURO_02         2005-05-01          ISAAC    RAMIREZ       NaN   
26           SEGURO_03         2005-05-01          JORGE     ROBLES       NaN   
27           SEGURO_12         2006-05-01         RAQUEL  DOMINGUEZ       NaN   
28           SEGURO_04         1993-05-01           IVAN     GARCIA       NaN   
30           SEGURO_10         2005-05-01        ROSARIO     ORTEGA       NaN   
32           SEGURO_03         2006-05-01    JUAN CARLOS     GARCIA       NaN   
33           SEGURO_07         2006-05-01         SERGIO    CARDONA       NaN   
34           SEGURO_01         2006-05-01           SARA     LOZANO       NaN   
35           SEGURO_04         2007-05-01          PEDRO    RAMIREZ       NaN   

0   NAC_FECHA    NUM_ID TIPO_ID DOM_SERVICIO_PROVINCIA  DOM_SERVICIO_CP  \
10 1996-04-20       NaN     DNI                GRANADA                8   
11 1991-09-19       NaN     DNI                GRANADA                8   
13 1992-05-19  45390239     DNI                GRANADA                8   
14 1993-02-02  84747702     DNI  MADRID (COMUNIDAD DE)               28   
17 1997-02-04  85022402     DNI                GRANADA                8   
18 1992-05-25  82675376     DNI                GRANADA                8   
19 1995-01-25  40546977     DNI                GRANADA                8   
20 1993-06-19  39621305     DNI                GRANADA                8   
21 1996-05-08  52910253     DNI                 MÁLAGA                8   
22 1996-06-22  26361148     DNI                GRANADA                8   
23 1993-12-08  63888659     DNI                GRANADA                8   
25 1992-11-12  17961949     DNI                GRANADA                8   
26 1995-01-18  77394278     DNI                GRANADA                8   
27 1995-07-09  50525577     DNI                GRANADA                8   
28 1994-04-14  37122848     DNI                CÓRDOBA               14   
30 1993-04-04  41624754     DNI                GRANADA                8   
32 1992-01-10  20039527     DNI                GRANADA                8   
33 1997-02-23  89408635     DNI                 MÁLAGA                8   
34 1996-08-11  75987051     DNI                GRANADA                8   
35 1994-09-08  58628967     DNI                GRANADA                8   

0  DOM_PAIS      TELF1 TELF2                                   EMAIL cliente  
10       ES  685918989   NaN          jesus.hernandez.diaz@gmail.com     NaN  
11       ES  673418203   NaN           carlos.torres.parra@gmail.com     NaN  
13       ES  695412798   NaN        daniel.piñeiro.herrera@gmail.com     NaN  
14       ES  645347696   NaN           rafael.gonzalez.mas@gmail.com     NaN  
17       ES  684918341   NaN        sergi.quesada.gonzalez@gmail.com     NaN  
18       ES  617935430   NaN      santiago.ramirez.navarro@gmail.com     NaN  
19       ES  664226881   NaN           ivan.robles.sanchez@gmail.com     NaN  
20       ES  613405794   NaN       amparo.dominguez.moreno@gmail.com     NaN  
21       ES

In [29]:
# There is no duplicate email.
df_email = df[df['EMAIL'].notnull()]
df_email[df_email.EMAIL.duplicated()]

Empty DataFrame
Columns: [Servicio contratado, FECHA_CONTRATACION, NOMBRE, APELLIDO1, APELLIDO2, NAC_FECHA, NUM_ID, TIPO_ID, DOM_SERVICIO_PROVINCIA, DOM_SERVICIO_CP, DOM_PAIS, TELF1, TELF2, EMAIL, cliente]
Index: []

In [30]:
# There is no duplicate tlf1.
df_tlf = df[df['TELF1'].notnull()]
df_tlf[df_tlf.TELF1.duplicated()]

Empty DataFrame
Columns: [Servicio contratado, FECHA_CONTRATACION, NOMBRE, APELLIDO1, APELLIDO2, NAC_FECHA, NUM_ID, TIPO_ID, DOM_SERVICIO_PROVINCIA, DOM_SERVICIO_CP, DOM_PAIS, TELF1, TELF2, EMAIL, cliente]
Index: []

In [31]:
# There is no duplicate tlf2.
df_tlf = df[df['TELF2'].notnull()]
df_tlf[df_tlf.TELF2.duplicated()]

Empty DataFrame
Columns: [Servicio contratado, FECHA_CONTRATACION, NOMBRE, APELLIDO1, APELLIDO2, NAC_FECHA, NUM_ID, TIPO_ID, DOM_SERVICIO_PROVINCIA, DOM_SERVICIO_CP, DOM_PAIS, TELF1, TELF2, EMAIL, cliente]
Index: []